src/GridView/HierarchyStepper.js

```diff
@@ -1,66 +1,60 @@
-import React, { Component } from 'react';
+import React from 'react';
 import { Step, Stepper, StepLabel } from 'material-ui/Stepper';
+import {connect} from "react-refetch";
+import ConnectionComponent from "./../Utils/ConnectionComponent.js";
 import { css } from 'aphrodite';
 import AppStyles from "./../AppStyles";
+import config from "./../config/config";
 
-class HierarchyStepper extends Component {
+
+class HierarchyStepper extends ConnectionComponent {
 	constructor() {
 		super();
-		this.entityTypes = [];
 		this.state = {
-			activeStep: 0,
-			highlitedEntityTypes: []
-		}
+            activeStep: 0,
+            hierarchy: [],
+		};
+		this.fetchParentEntity = this.fetchParentEntity.bind(this);
 	}
 
 	componentDidMount() {
-		this.updateEntityTypes();
-		const highlitedEntityTypes = this.getHighlitedEntityTypes(this.props.currentEntityTypeId)
 		this.setState({
-			activeStep: highlitedEntityTypes.length,
-			highlitedEntityTypes: highlitedEntityTypes
-		});
-
-	}
-
-	updateEntityTypes() {
-		this.props.hierarchy.forEach((hierarchyLayer) => {
-			this.entityTypes = this.entityTypes.concat(hierarchyLayer);
-		})
-	}
-
-	getEntityType(EntityTypeId) {
-		return (this.entityTypes.find((entityType) => {
-			return (entityType.Id === EntityTypeId);
-		}));
+			hierarchy: [this.props.getChildEntityTypes(this.props.currentEntity.TypeId, this.props.hierarchy)]
+		}, () => {this.fetchParentEntity(this.props.currentEntity)});
 	}
 
-	getHighlitedEntityTypes(currentEntityTypeId) {
-		const currentEntityType = this.getEntityType(currentEntityTypeId);
-
-		if (currentEntityTypeId === -1) {
-			return [];
-		}
-		else if (currentEntityType.ParentId === -1) {
-			return [currentEntityType];
-		}
-		else {
-			const parentEntityType = this.getEntityType(currentEntityType.ParentId);
-			return this.getHighlitedEntityTypes(parentEntityType.Id).concat([currentEntityType]);
+	fetchParentEntity(entity) {
+		if (entity.Id !== -1) {
+            this.setState({
+                activeStep: this.state.activeStep + 1,
+                hierarchy: [[entity]].concat(this.state.hierarchy),
+            });
 		}
+        if (entity.ParentId !== -1) {
+            const entityFetch = {
+                entityId: entity.ParentId,
+                successCallback: this.fetchParentEntity,
+			};
+            this.props.lazyEntityFetch(entityFetch);
+        }
 	}
 
-	displayStepLabel(entityType, index) {
-		if(this.state.highlitedEntityTypes.includes(entityType)) {
-			return (
-				<StepLabel
-					key={index}
-					completed={true}>
-					{entityType.Name}
-				</StepLabel>);
+	displayStepLabel(hierarchyLayerInstance, key) {
+		if (hierarchyLayerInstance.TypeId >= 0) {
+            return(
+				<StepLabel key={key}>
+					<a href={`/grid/${hierarchyLayerInstance.Id}`}>
+						{this.props.getEntityType(hierarchyLayerInstance, this.props.hierarchy)[0].Name}: {hierarchyLayerInstance.Name}
```

Comment:
Looks a bit too long for me.
Maybe the `this.props.getEnt....Name` can be extracted to a variable.